packages/wallets/backpack/src/adapter.ts

```diff
@@ -0,0 +1,259 @@
+    }
+
+    async connect(): Promise<void> {
+        try {
+            if (this.connected || this.connecting) return;
+            if (this._readyState !== WalletReadyState.Installed) throw new WalletNotReadyError();
+            this._connecting = true;
+
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            const wallet = window!.backpack!;
+
+            if (!wallet.isConnected) {
+                try {
+                    await new Promise<void>((resolve, reject) => {
+                        const handleConnect = () => {
+                            wallet.off('connect', handleConnect);
+                            resolve();
+                        };
+
+                        // Backpack emits a connect event when it is ready
+                        wallet.on('connect', handleConnect);
+
+                        wallet.connect().catch(() => {
+                            wallet.off('connect', handleConnect);
+                            reject(new WalletWindowClosedError());
+                        });
+                    });
+                } catch (error: any) {
+                    if (error instanceof WalletError) throw error;
+                    throw new WalletConnectionError(error?.message, error);
+                }
+            }
+
+            if (!wallet.publicKey) throw new WalletAccountError();
+
+            let publicKey: PublicKey;
+            try {
+                publicKey = new PublicKey(wallet.publicKey.toBytes());
+            } catch (error: any) {
+                throw new WalletPublicKeyError(error?.message, error);
+            }
+            wallet.on('disconnect', this._disconnected);
+
+            this._wallet = wallet;
+            this._publicKey = publicKey;
+
+            this.emit('connect', publicKey);
+        } catch (error: any) {
+            this.emit('error', error);
+            throw error;
+        } finally {
+            this._connecting = false;
+        }
+    }
+
+    async disconnect(): Promise<void> {
+        const wallet = this._wallet;
+        if (wallet) {
+            wallet.off('disconnect', this._disconnected);
+
+            this._wallet = null;
+            this._publicKey = null;
+
+            try {
+                await wallet.disconnect();
+            } catch (error: any) {
+                this.emit('error', new WalletDisconnectionError(error?.message, error));
+            }
+        }
+
+        this.emit('disconnect');
+    }
+
+    async sendTransaction(
+        transaction: Transaction,
+        connection?: Connection,
+        options?: SendTransactionOptions
+    ): Promise<TransactionSignature> {
+        try {
+            const wallet = this._wallet;
+            if (!wallet) throw new WalletNotConnectedError();
+
+            const { signers, ...sendOptions } = options ? options : { signers: undefined };
+
+            const resp = await wallet.send(transaction, signers, sendOptions, connection);
```

Comment:
Just to clarify, is `resp` a signature or an object containing a signature field like many of the wallets do?

Reply:
It's a signature. I'd prefer to leave it that way due to the Anchor interface https://github.com/coral-xyz/anchor/blob/master/ts/src/provider.ts#L24-L28. 

And now that I see that, I realize we should change the `| null` return values in the connect, send, sign, etc apis to throw to match anchor.

Reply:
Ah, all good! I didn't mean to suggest changing it, was only concerned that maybe it was a response object rather than a sig from the varname.